x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/update_rules_route.test.ts

```diff
@@ -4,163 +4,119 @@
-      clients.alertsClient.get.mockResolvedValue(getResult());
-      clients.actionsClient.update.mockResolvedValue(updateActionResult());
-      clients.alertsClient.update.mockResolvedValue(getResult());
-      clients.savedObjectsClient.find.mockResolvedValue(getFindResultStatus());
-      const { statusCode } = await server.inject(getUpdateRequest());
-      expect(statusCode).toBe(200);
+      const response = await server.inject(getUpdateRequest(), context);
+      expect(response.status).toEqual(200);
     });
 
     test('returns 404 when updating a single rule that does not exist', async () => {
-      clients.alertsClient.find.mockResolvedValue(getFindResult());
-      clients.alertsClient.get.mockResolvedValue(getResult());
-      clients.actionsClient.update.mockResolvedValue(updateActionResult());
-      clients.alertsClient.update.mockResolvedValue(getResult());
-      clients.savedObjectsClient.find.mockResolvedValue(getFindResultStatus());
-      const { statusCode } = await server.inject(getUpdateRequest());
-      expect(statusCode).toBe(404);
+      clients.alertsClient.find.mockResolvedValue(getEmptyFindResult());
+      const response = await server.inject(getUpdateRequest(), context);
+
+      expect(response.status).toEqual(404);
+      expect(response.body).toEqual({
+        message: expect.stringMatching(/rule_id.*not found/),
+        status_code: 404,
+      });
     });
 
     test('returns 404 if alertClient is not available on the route', async () => {
-      getClients.mockResolvedValue(omit('alertsClient', clients));
-      const { route, inject } = createMockServer();
-      updateRulesRoute(route, config, getClients);
-      const { statusCode } = await inject(getUpdateRequest());
-      expect(statusCode).toBe(404);
+      context.alerting.getAlertsClient = jest.fn();
+      const response = await server.inject(getUpdateRequest(), context);
+
+      expect(response.status).toEqual(404);
+      expect(response.body).toEqual({ message: 'Not Found', status_code: 404 });
     });
 
-    test('returns 500 when transform fails', async () => {
-      clients.alertsClient.find.mockResolvedValue(getFindResultWithSingleHit());
-      clients.alertsClient.get.mockResolvedValue(getResult());
-      clients.savedObjectsClient.find.mockResolvedValue(getFindResultStatus());
-      jest.spyOn(utils, 'transform').mockReturnValue(null);
-      const { payload, statusCode } = await server.inject(getUpdateRequest());
-      expect(JSON.parse(payload).message).toBe('Internal error transforming');
-      expect(statusCode).toBe(500);
+    test('returns error when updating non-rule', async () => {
+      clients.alertsClient.find.mockResolvedValue(nonRuleFindResult());
+      const response = await server.inject(getUpdateRequest(), context);
+
+      expect(response.status).toEqual(404);
+      expect(response.body).toEqual({
+        message: expect.stringMatching(/rule_id.*not found/),
```

Comment:
I'd be careful about changing exact matches in tests with regular expressions as you can get false positives. I am not asking you to change these but with regular expressions you get less safety as a trade off.

For example if your error message handler down the road begins returning:

```ts
"rule_id: \"undefined\" not found"
```

instead of:

```ts
"rule_id: \"rule-1\" not found"
```

These types of tests would still end up passing.

They would also pass if the colon was missing since you started the regular expression at `rule_id.*` rather than at the colon so that would become a false positive if that changes as well.

A more exact one would be this:

```ts
/rule_id:\s".*"\snot found/
```

But now your test is a bit trickier to read. So no changes needed, just my opinion about the trade offs. This looks like a really good PR so I don't want to be too picky about some of the details.

Reply:
https://github.com/elastic/kibana/pull/58292/commits/5f37eba891576fab388ee977ca237f75f4e343ef